Whiskey/Tasks/Invoke-WhiskeyNpmScript.ps1

```diff
@@ -0,0 +1,164 @@
+
+        if (Test-Path $nodeModulesPath -PathType Container)
+        {
+            $outputDirectory = Join-Path -path $TaskContext.BuildRoot -ChildPath '.output' 
+            $emptyDir = New-Item -Name 'TempEmptyDir' -Path $outputDirectory -ItemType 'Directory'
+
+            Write-Timing -Message ('Emptying {0}' -f $nodeModulesPath)
+            Invoke-WhiskeyRobocopy -Source $emptyDir -Destination $nodeModulesPath | Write-Debug
+            Write-Timing -Message ('COMPLETE')
+
+            Remove-Item -Path $emptyDir
+            Remove-Item -Path $nodeModulesPath
+        }
+        return
+    }
+
+    $npmRegistryUri = $TaskParameter['NpmRegistryUri']
+    if (-not $NpmRegistryUri) 
+    {
+        Stop-WhiskeyTask -TaskContext $TaskContext -Message 'Property ''NpmRegistryUri'' is mandatory. It should be the URI to the registry from which Node.js packages should be downloaded, e.g.,
+        
+        BuildTasks:
+        - NpmScript:
+            NpmRegistryUri: https://registry.npmjs.org/
+
+        '
+    }
+
+    $npmScripts = $TaskParameter['Script']
+    if (-not $npmScripts)
+    {
+        Stop-WhiskeyTask -TaskContext $TaskContext -Message 'Property ''Script'' is mandatory. It should be a list of one or more npm scripts to run during your build, e.g.,
+
+        BuildTasks:
+        - NpmScript:
+            Script:
+            - build
+            - test
+
+        '
+    }
+
+    $workingDirectory = $TaskContext.BuildRoot
+    if ($TaskParameter['WorkingDirectory'])
+    {
+        $workingDirectory = $TaskParameter['WorkingDirectory'] | Resolve-WhiskeyTaskPath -TaskContext $TaskContext -PropertyName 'WorkingDirectory'
+    }
+
+    Push-Location -Path $workingDirectory
+    try
+    {
+        $noColorArg = ''
+        if (($TaskContext.ByBuildServer) -or ($Host.Name -ne 'ConsoleHost'))
+        {
+            $noColorArg = '--no-color'
+        }
+
+        Write-Timing -Message 'Installing Node.js modules.'
+        Invoke-WhiskeyNpmCommand -NpmCommand 'install' -Argument @('--production=false', $noColorArg) -ApplicationRoot $workingDirectory -RegistryUri $NpmRegistryUri -ForDeveloper:$TaskContext.ByDeveloper
+        Write-Timing -Message 'COMPLETE'
+
+        if ($LASTEXITCODE -ne 0)
+        {
+            Stop-WhiskeyTask -TaskContext $TaskContext -Message ('NPM command ''npm install'' failed with exit code ''{0}''.' -f $LASTEXITCODE)
+        }
+
+        if ($TaskContext.ShouldInitialize())
+        {
+            Write-Timing -Message 'Initialization complete.'
+            return
+        }
+
+        foreach ($script in $npmScripts)
+        {
+            Write-Timing -Message ('Running script ''{0}''.' -f $script)
+            Invoke-WhiskeyNpmCommand -NpmCommand 'run' -Argument @($script, '--scripts-prepend-node-path=auto', $noColorArg) -ApplicationRoot $workingDirectory -RegistryUri $NpmRegistryUri -ForDeveloper:$TaskContext.ByDeveloper

```

Comment:
Should the `--scripts-prepend-node-path=auto` be passed to every NPM command? Should this get moved into `Invoke-WhiskeyNpmCommand`? Same with the `$noColorArg`?

Reply:
It definitely should be moved into `Invoke-WhiskeyNpmCommand`.